### DISCUSSION TITLE
Redis key prefix

### DISCUSSION THREAD

**Discussion by monwolf** [renovatebot/renovate#26393] [Request Help] [answered]
### How are you running Renovate?

None

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.91

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### What is your question?

We use renovate in a self-hosted environment on top of Nomad, using a shared Redis for multiple applications.  Each application can only write in its custom space of keys.  For example,  "app1" only can access and manage keys with the prefix "app1/*"

We want to enable the caching mechanism with Redis but I don't see any way to use custom prefixes in Redis keys:
![image](https://github.com/renovatebot/renovate/assets/1116133/66c2201f-c10a-44c0-94bc-a2bc41799772)

Is there any way to achieve it? 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Please provide more references, and ideally a suggestion on how configuration would look like. For example, it's not something you'd add to the URL itself?

**Reply by monwolf** [reply]
This is something we setup in the redis configuration, the user app1 just can write in app1 namespace:

```
user app1 on > passwd1 ~app1:* +@all -@dangerous +info
```

And then we force all applications to write in this namespace.

Some software implementations allow setting up the cache prefix with a custom property. For example, springboot enable this config withe the parameter _spring.cache.redis.key-prefix_, in laravel it uses the variable _REDIS_PREFIX_, ...

**Reply by rarkins** [reply]
Please locate some official documentation/references we can correlate this to

**Reply by monwolf** [reply]
I made a mistake, REDIS_PREFIX is not standard is one thing we added to our apps, to set it up you have the config "prefix" inside de redis object
https://laravel.com/docs/10.x/redis#phpredis
And then we added to our app as follows:
```
'prefix' => env('REDIS_PREFIX', 'default_prefix'),
```


From springboot's docs:

https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html#application-properties.cache.spring.cache.redis.key-prefix

https://docs.spring.io/spring-data/redis/reference/redis/redis-cache.html

**Reply by rarkins** [reply, marked as answer]
Converted to feature request issue: https://github.com/renovatebot/renovate/issues/26394

PRs welcome
